src/GeositeFramework/js/Map.js

```diff
@@ -206,33 +204,18 @@ require(['use!Geosite',
             id = 'legend-container-' + mapNumber,
             legend = new Legend(regionData, id);
 
-        var getServiceLegend = (function() {
-            var cache = {};
-            // Memoize so only 1 request is generated per service.
-            var fetch = _.memoize(function(url) {
-                xhr(url, {
-                        query: 'f=json',
-                        method: 'GET',
-                        handleAs: 'json',
-                        headers: {
-                            // Fixes: "Request header field X-Requested-With is not allowed by Access-Control-Allow-Headers in preflight response."
-                            'X-Requested-With': null
-                        }
-                    })
-                    .then(function(data) {
-                        cache[url] = data.layers;
-                    })
-                    .then(redraw);
-            });
-            return function(serviceUrl) {
-                var url = serviceUrl + '/legend';
-                if (cache[url]) {
-                    return cache[url];
-                }
-                fetch(url);
-                return null;
-            };
-        }());
+        var redraw = function() {
+            legend.render(getVisibleLayers());
+        };
+
+        function getServiceLegend(service) {
+            var legendUrl = service.url + '/legend',
+                data = ajaxUtil.get(legendUrl);
+            if (ajaxUtil.shouldFetch(legendUrl)) {
```

Comment:
It's not totally clear to me here why `get` is called before the data is fetched. `data` will be populated by `fetch` so that when this function returns it's not undefined if the cache was empty initially?